db/db_impl_compaction_flush.cc

```diff
@@ -1142,6 +1158,54 @@ Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
   return s;
 }
 
+Status DBImpl::FlushMemTablesInternal(const FlushOptions& flush_options,
+    FlushReason flush_reason, bool writes_stopped) {
+  Status s;
+  std::vector<ColumnFamilyData*> cfds;
+  std::vector<uint64_t> nums;
+  std::vector<uint64_t*> flush_memtable_ids;
+  {
+    WriteContext context;
+    InstrumentedMutexLock guard_lock(&mutex_);
+
+    WriteThread::Writer w;
+    if (!writes_stopped) {
+      write_thread_.EnterUnbatched(&w, &mutex_);
+    }
+
+    for (auto cfd : *versions_->GetColumnFamilySet()) {
+      if (cfd->IsDropped() ||
+          (cfd->imm()->NumNotFlushed() == 0 && cfd->mem()->IsEmpty() &&
+          cached_recoverable_state_empty_.load())) {
+        continue;
+      }
+      // SwitchMemtable will release and re-acquire mutex during execution
+      s = SwitchMemtable(cfd, &context);
+      if (!s.ok()) {
+        break;
+      }
+      cfds.push_back(cfd);
+      uint64_t flush_memtable_id = cfd->imm()->GetLatestMemTableID();
+      nums.push_back(flush_memtable_id);
+      flush_memtable_ids.push_back(&nums[nums.size() - 1]);
+      cfd->imm()->FlushRequested();
+      // cfd will be refcounted.
+      SchedulePendingFlush(cfd, flush_reason);
```

Comment:
The flushes to different column families are still scheduled independently. They aren't necessarily atomic, as they are not writing to the manifest atomically.

Reply:
In the foreground, all column families are scheduled for flushing if `atomic_flush=true`. In this sense, column families are not scheduled independently.